### PR TITLE
Adding cs question to wesite

### PR DESCRIPTION
**Pull request by deep-yad**
Added new cs question in the repo

## Checklist

- [ x] I have read and followed the [contribution guidelines](https://github.com/freeCodeCamp/Developer_Quiz_Site/blob/main/CONTRIBUTING.md).
- [ x] I have read through the [Code of Conduct](https://github.com/freeCodeCamp/Developer_Quiz_Site/blob/main/CODE_OF_CONDUCT.md) and agree to abide by the rules.
- [x ] This PR is for one of the available issues and is not a PR for an issue already assigned to someone else.
- [x ] My PR title has a short descriptive name so the maintainers can get an idea of what the PR is about.
- [ x] I have provided a summary of my changes.

closes #XXXXX

### CODE REVIEW

File: src/data/general-cs-quiz.ts [hunk 1]
```diff
@@ -1,4 +1,27 @@
 const generalCSQuiz = [
+  {
+    Question:
+      "Which sorting algorithm has an average and worst-case time complexity of O(nlogn)?",
```
Comment (jdwilkin4) [inline src/data/general-cs-quiz.ts:4]: worst case would be O(n^2) depending on the selection of the pivot element
so the question should be updated to reflect that

```suggestion
      "Which sorting algorithm has an average time complexity of O(nlogn)?",
```

File: src/data/general-cs-quiz.ts [hunk 2]
```diff
@@ -1,4 +1,27 @@
...
+    Answer: "Quick sort",
+    Distractor1: "Bubble sort",
+    Distractor2: "Insertion sort",
+    Distractor3: "Selection sort",
+    Explanation:
+      "Quick sort has an average and worst-case time complexity of O(n log n), making it efficient for sorting large datasets.",
+    Link: "https://www.freecodecamp.org/news/sorting-algorithms-explained-with-examples-in-python-java-and-c/#classification-of-a-sorting-algorithm"
```
Comment (jdwilkin4) [inline src/data/general-cs-quiz.ts:11]:
```suggestion
    Link: "https://www.freecodecamp.org/news/sorting-algorithms-explained-with-examples-in-python-java-and-c/#quick-sort"
```

File: src/data/general-cs-quiz.ts [hunk 3]
```diff
@@ -1,4 +1,27 @@
 const generalCSQuiz = [
+  {
+    Question:
+      "Which sorting algorithm has an average and worst-case time complexity of O(nlogn)?",
+    Answer: "Quick sort",
+    Distractor1: "Bubble sort",
+    Distractor2: "Insertion sort",
+    Distractor3: "Selection sort",
+    Explanation:
+      "Quick sort has an average and worst-case time complexity of O(n log n), making it efficient for sorting large datasets.",
```
Comment (jdwilkin4) [inline src/data/general-cs-quiz.ts:10]:
```suggestion
      "Quick sort has an average time complexity of O(n log n), making it efficient for sorting large datasets.",
```

File: src/data/general-cs-quiz.ts [hunk 4]
```diff
@@ -1,4 +1,27 @@
...
+    Link: "https://www.freecodecamp.org/news/sorting-algorithms-explained-with-examples-in-python-java-and-c/#classification-of-a-sorting-algorithm"
+  },
+  {
+    Question:
+      " In a graph, which data structure is best suited for finding the shortest path between two vertices?",
```
Comment (jdwilkin4) [inline src/data/general-cs-quiz.ts:15]: remove extra spaces

```suggestion
      "In a graph, which data structure is best suited for finding the shortest path between two vertices?",
```

File: src/data/general-cs-quiz.ts [hunk 5]
```diff
@@ -1,4 +1,27 @@
...
+    Answer: ". Priority Queue",
```
Comment (jdwilkin4) [inline src/data/general-cs-quiz.ts:16]: remove spaces and period

```suggestion
    Answer: "Priority Queue",
```

File: src/data/general-cs-quiz.ts [hunk 6]
```diff
@@ -1,4 +1,27 @@
...
+    Distractor1: "Stack",
+    Distractor2: "Queue",
```
Comment (jdwilkin4) [inline src/data/general-cs-quiz.ts:18]:
```suggestion
    Distractor2: "string",
```

File: src/data/general-cs-quiz.ts [hunk 7]
```diff
@@ -1,4 +1,27 @@
...
+    Distractor3: "Heap",
+    Explanation:
+      "Dijkstra's algorithm, which is used to find the shortest path in a graph, typically employs a            priority queue to efficiently select the next vertex with the smallest distance.",
```
Comment (jdwilkin4) [inline src/data/general-cs-quiz.ts:21]: remove unnecessary spaces 

```suggestion
      "Dijkstra's algorithm, which is used to find the shortest path in a graph, typically employs a priority queue to efficiently select the next vertex with the smallest distance.",
```

File: src/data/general-cs-quiz.ts [hunk 8]
```diff
@@ -1670,15 +1694,16 @@ const generalCSQuiz = [
...
+    Question: "What is the use of a priority queue?",
+    Answer:
+      "A priority queue manages elements based on priority, ensuring higher priority items are processed first.",
+    Distractor1: "It's a queue that organizes elements by their value.",
+    Distractor2: "It's a fixed-size queue.",
+    Distractor3: "It only accepts elements of the same type.",
+    Explanation:
+      "A priority queue is essential in scenarios where certain tasks need to be handled in order of importance or urgency.",
+    Link: "https://www.freecodecamp.org/news/priority-queue-implementation-in-java/"
+  }
```
Comment (jdwilkin4) [inline src/data/general-cs-quiz.ts:1706]: this question already exists in the file.
Please remove it

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]
Thank you for your contribution!

Please review the requested changes.
Once those changes have been implemented, then we can merge your PR into main. 👍

File: src/data/general-cs-quiz.ts (hunk 8 above)
Comment (jdwilkin4) [reply src/data/general-cs-quiz.ts:1706]: This still need to be resolved

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]

File: src/data/general-cs-quiz.ts (hunk 8 above)
Comment (deep-yad) [reply src/data/general-cs-quiz.ts:1706]: There is only one question "what is use of Priority queues?"

Comment (deep-yad) [reply src/data/general-cs-quiz.ts:1706]: Do you want me to remove it?

Comment (jdwilkin4) [reply src/data/general-cs-quiz.ts:1706]: yes, you need to remove the priority queue question because we already have a similar one in the file

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]

File: src/data/general-cs-quiz.ts (hunk 8 above)
Comment (jdwilkin4) [reply src/data/general-cs-quiz.ts:1706]: ok, I see what happened.
The change was a formatting change by the prettier formatting tool.
We can leave it then.

**Review by jdwilkin4 — APPROVED** [review]
Congrats on another contribution to our repository. 🎉
We look forward to future contributions. 👍

**Pull request merged**